pkg/binary/envoy/controlplane/istio_test.go

```diff
@@ -38,39 +37,57 @@ func TestMain(m *testing.M) {
 	envoytest.FetchEnvoyAndRun(m)
 }
 
+// TestConnectsToMockPilotAsAGateway ensures Istio bootstraps Envoy configuration, and that configuration in
+// "networking.istio.io/v1alpha3" rules end up pushed back to Envoy via XDS.
+//
+// Technically what happens is bootstrap results in a "xds-grpc" cluster pointed at Pilot's gRPC address. During
+// bootstrap, Envoy invokes "/envoy.service.discovery.v3.AggregatedDiscoveryService/StreamAggregatedResources", which
+// reads configuration in pilot (from testdata/configs.yaml). This test passes when that configuration merges with the
+// bootstrap configuration generated by Istio.
 func TestConnectsToMockPilotAsAGateway(t *testing.T) {
-	pilotGrpcAddr, stopPilot := requireMockPilot(t, "testdata")
+	// only includes configs.yaml, which includes egress rules for googleapis
+	pilotServer, stopPilot := requireMockPilot(t, "testdata")
 	defer stopPilot()
 
+	pilotGrpc := pilotServer.GRPCListener.Addr().String()
 	runtime, err := envoy.NewRuntime(
 		func(r *envoy.Runtime) {
 			r.Config.Mode = envoy.ParseMode("loadbalancer")
-			r.Config.XDSAddress = pilotGrpcAddr
-			r.Config.IPAddresses = []string{"1.1.1.1"}
+			// Choosing a random port doesn't help because envoy_bootstrap.json hard-codes 15020 15021 and 15090
```

Comment:
ps anyone in istio care about this problem in the bootstrap templating? doesn't seem to make much sense to only template the admin port while the others are hard-coded.. that or I am missing something. At any rate, using port 0 and reading back the value of `--admin-address-path` would be ideal.

Reply:
yeah, i think it's just because no one complains about it so far...

Reply:
impact for us is we can't do parallel testing because of this. good news is we only have one istio test :D